core/services/webhook/delegate.go

```diff
@@ -15,27 +15,63 @@ import (
 
 type (
 	Delegate struct {
-		webhookJobRunner *webhookJobRunner
+		webhookJobRunner         *webhookJobRunner
+		externalInitiatorManager ExternalInitiatorManager
 	}
 
 	JobRunner interface {
 		RunJob(ctx context.Context, jobUUID models.JobID, pipelineInput interface{}, meta pipeline.JSONSerializable) (int64, error)
 	}
 )
 
-func NewDelegate(runner pipeline.Runner) *Delegate {
+var _ job.Delegate = (*Delegate)(nil)
+
+func NewDelegate(runner pipeline.Runner, externalInitiatorManager ExternalInitiatorManager) *Delegate {
 	return &Delegate{
+		externalInitiatorManager: externalInitiatorManager,
 		webhookJobRunner: &webhookJobRunner{
 			specsByUUID: make(map[models.JobID]registeredJob),
 			runner:      runner,
 		},
 	}
 }
 
+func (d *Delegate) WebhookJobRunner() JobRunner {
+	return d.webhookJobRunner
+}
+
 func (d *Delegate) JobType() job.Type {
 	return job.Webhook
 }
 
+func (d *Delegate) OnJobCreated(spec job.Job) {
+	if !spec.WebhookSpec.ExternalInitiatorName.IsZero() {
```

Comment:
You could also use `if spec.WebhookSpec.ExternalInitiatorName.Valid {`

Reply:
Tomato